File: backend/app/api/assistant.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.deps import require_permission
from app.core.response import success
from app.db.session import get_db
from app.models.user import User
from app.schemas.assistant import AssistantChatRequest
from app.services.assistant_service import assistant_status, chat_with_assistant
from app.services.log_service import create_log

router = APIRouter(prefix="/assistant", tags=["assistant"])


@router.get("/status")
def get_assistant_status(_: User = Depends(require_permission("assistant:use"))):
    return success(assistant_status())


@router.post("/chat")
def chat_api(payload: AssistantChatRequest, db: Session = Depends(get_db), current_user: User = Depends(require_permission("assistant:use"))):
    data = chat_with_assistant(db, current_user, payload.question)
    create_log(db, "assistant", "AI assistant question completed", module="assistant", user_id=current_user.id)
    return success(data)

File: frontend/src/api/assistant.ts
import { request, unwrap } from './request'

export interface AssistantStatus {
  configured: boolean
  model: string
  base_url?: string
}

export interface AssistantChatResponse {
  answer: string
  model: string
  configured: boolean
}

export function getAssistantStatus() {
  return unwrap<AssistantStatus>(request.get('/assistant/status'))
}

export function askAssistant(question: string) {
  return unwrap<AssistantChatResponse>(request.post('/assistant/chat', { question }))
}
